pkg/common/common.go

```diff
@@ -206,17 +211,34 @@ func AppendTTMapAsCopy[T comparable](add ...map[T]T) map[T]T {
 	return base
 }
 
-// CCOWorkflow checks if the AWS STS secret is to be obtained from Cloud Credentials Operator (CCO)
+// CCOWorkflow checks if the AWS STS secret or GCP WIF is to be obtained from Cloud Credentials Operator (CCO)
 // if the user provides role ARN during installation then the ARN gets set as env var on operator deployment
-// during installation via OLM
+// during installation via OLM. Similarly, if the user provides AUDIENCE and SERVICE_ACCOUNT_EMAIL for GCP WIF,
+// these are set as env vars on the operator deployment.
 func CCOWorkflow() bool {
-	roleARN := os.Getenv("ROLEARN")
-	if len(roleARN) > 0 {
+	roleARN := os.Getenv(RoleARNEnvKey)
+	audience := os.Getenv(AudienceEnvKey)
+	serviceAccountEmail := os.Getenv(ServiceAccountEmailEnvKey)
+
+	if len(roleARN) > 0 || (len(audience) > 0 && len(serviceAccountEmail) > 0) {
```

Comment:
why not use `IsAWSSTS` and `IsGCPWIF` functions here instead?

Reply:
no no

something like this
```go
func CCOWorkflow() bool {
	return IsAWSSTS() && IsGCPWIF()
}
```

Reply:
But each cloud has different secret pattern

Reply:
All this does is create secret.. so maybe refactor so that its CreateSecretForCCOAWSSTS, and CreateSecretForCCOGCPWIF and within each check if env exists to do so.

Reply:
That way we don't have to 
```go
func CCOWorkflow() bool {
	return IsAWSSTS() && IsGCPWIF()
}
// then later also do
if IsAWSSTS() { doSomethingForAWS() }
```